### PR TITLE
perf: add experimental support for `importModule`

### PR DESCRIPTION
**Pull request by sokra**
This replaces the child compiler with a
lightweight execution of modules within
the module graph of the original compilation

To try the new approach.

* Use webpack >= 5.32.0
* `experimentalUseImportModule: true` in `MiniCssPlugin` options
* Avoid `publicPath` in `MiniCssPlugin` options, that doesn't work yet
* Set a static `output.publicPath` in webpack options

This PR contains a:

- [x] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

child compiler take up too much memory and performance

### Breaking Changes

none

### Additional Info

Not all tests are passing yet with the new approach.
There is nothing to do in mini-css-extract-plugin to fix this.
Currently I think that are all webpack problems that I will fix eventually.
Biggest problem currently is that it breaks with HMR.

### CODE REVIEW

**Comment by sokra** [comment]
@vyobukhov https://github.com/vyobukhov/mini-css-memory-issues builds within `--max-old-space-size=512` with this PR.

File: src/loader.js [hunk 1]
```diff
@@ -67,6 +187,40 @@ export function pitch(request) {
...
+  if (optionsFromPlugin.experimentalUseImportModule) {
+    if (!this.importModule) {
+      callback(
+        new Error(
+          "You are using experimentalUseImportModule but 'this.importModule' is not available in loader context. You need to have at least webpack 5.33.2."
+        )
+      );
+      return;
+    }
+    console.log(publicPath);
```
Comment (alexander-akait) [inline src/loader.js:199]: I think we need avoid it :smile:

**Review by alexander-akait — COMMENTED** [review]
Need remove `console.log` and I think we can merge and release it

File: src/loader.js (hunk 1 above)
Comment (sokra) [reply src/loader.js:199]: Sure. Fixed

**Comment by sokra** [comment]
🚀

**Review by alexander-akait — APPROVED** [review]

**Comment by alexander-akait** [comment]
I will do release tomorrow :rocket:

**Pull request merged**